fpdf/image_parsing.py

```diff
@@ -172,15 +172,17 @@ def _to_zdata(img, remove_slice=None, select_slice=None):
         data = data[select_slice]
     # Left-padding every row with a single zero:
     if img.mode == "1":
-        loop_incr = ceil(img.size[0] / 8) + 1
+        row_size = ceil(img.size[0] / 8)
     else:
         channels_count = len(data) // (img.size[0] * img.size[1])
-        loop_incr = img.size[0] * channels_count + 1
-    i = 0
-    while i < len(data):
-        data[i:i] = b"\0"
-        i += loop_incr
-    return zlib.compress(data)
+        row_size = img.size[0] * channels_count
+
+    data_with_padding = bytearray()
```

Comment:
👍 Nice! Well done!

Reply:
My changes have almost no effect on the speed of this test because this test works with small images (less than 100x100 pixels). At these sizes the asymptotics of O(N) is not much different from O(N^2) in comparison with other operations.

However, personally on my computer, comparable in power to Gitlab CI, it takes only 11.6 seconds to run. So it is probably possible to reduce the threshold from 40 seconds to 20 seconds.

Reply:
Alright, thank you for your feedback!

I'm going to merge this and see in a later PR if I can bump this threshold

Reply:
Seems like 30s is too low for GitHub Actions runners:
https://github.com/PyFPDF/fpdf2/pull/647

Nevermind 😄